pandas/core/indexes/datetimes.py

```diff
@@ -213,6 +213,18 @@ class DatetimeIndex(DatelikeOps, TimelikeOps, DatetimeIndexOpsMixin,
            Attempt to infer fall dst-transition hours based on order
     name : object
         Name to be stored in the index
+    dayfirst : bool, default False
+        If True, parse dates in `data` with the day first,
+        e.g. 10/11/12 is parsed as 2012-11-10.
+        Warning: dayfirst=True is not strict, but will prefer to parse
+        with day first (this is a known bug, based on dateutil behavior).
+    yearfirst : bool, default False
+        - If True parse dates in `data` with the year first,
+          e.g. 10/11/12 is parsed as 2010-11-12.
+        - If both dayfirst and yearfirst are True, yearfirst is preceded (same
+          as dateutil).
+        Warning: yearfirst=True is not strict, but will prefer to parse
+        with year first (this is a known bug, based on dateutil beahavior).
```

Comment:
Are dayfirst/yearfist docstring entries copied from somewhere?  Might go well in a shared_docs?

Reply:
Not exactly, but I'll try to do something with that, thanks.